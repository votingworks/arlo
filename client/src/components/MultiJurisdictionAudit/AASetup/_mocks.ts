import { ElementType } from '../../../types'
import { setupStages, stageTitles } from './index'
import { ISidebarMenuItem } from '../../Atoms/Sidebar'

const relativeStages = (
  stage: ElementType<typeof setupStages>,
  state: ISidebarMenuItem['state'] = 'live'
): {
  prevStage: ISidebarMenuItem
  nextStage: ISidebarMenuItem
  menuItems: ISidebarMenuItem[]
} => {
  const prevId = setupStages[setupStages.indexOf(stage) - 1]
  const prevStage: ISidebarMenuItem = {
    id: prevId,
    title: stageTitles[prevId],
    active: false,
    activate: jest.fn(),
    state,
  }
  const nextId = setupStages[setupStages.indexOf(stage) + 1]
  const nextStage: ISidebarMenuItem = {
    id: nextId,
    title: stageTitles[nextId],
    active: false,
    activate: jest.fn(),
    state,
  }
  const menuItems = setupStages.map((s: ElementType<typeof setupStages>) => ({
    id: s,
    title: stageTitles[s],
    active: s === stage,
    activate: jest.fn(),
    state,
  }))
  return { prevStage, nextStage, menuItems }
}

export default relativeStages
