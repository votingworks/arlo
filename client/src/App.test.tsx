import React from 'react'
import { screen, render } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'
import App from './App'
import { withMockFetch, renderWithRouter } from './components/testUtilities'
import { dummyBoards } from './components/DataEntry/_mocks'
import {
  auditSettings,
  manifestMocks,
  talliesMocks,
  cvrsMocks,
} from './components/MultiJurisdictionAudit/useSetupMenuItems/_mocks'
import {
  jaApiCalls,
  aaApiCalls,
} from './components/MultiJurisdictionAudit/_mocks'

jest.unmock('react-toastify')

const apiMocks = {
  failedAuth: {
    url: '/api/me',
    response: null,
  },
  abAuth: {
    url: '/api/me',
    response: { type: 'audit_board', ...dummyBoards()[1] },
  },
}

const renderView = (route: string) => renderWithRouter(<App />, { route })

describe('App', () => {
  describe('/', () => {
    it('renders unauthenticated properly', async () => {
      const expectedCalls = [apiMocks.failedAuth]
      await withMockFetch(expectedCalls, async () => {
        const { container } = renderView('/')
        await screen.findByAltText('Arlo, by VotingWorks')
        expect(container).toMatchSnapshot()
      })
    })

    it('renders ja logged in properly', async () => {
      const expectedCalls = [jaApiCalls.getUser]
      await withMockFetch(expectedCalls, async () => {
        const { container } = renderView('/')
        await screen.findByAltText('Arlo, by VotingWorks')
        expect(container).toMatchSnapshot()
      })
    })

    it('renders aa logged in properly', async () => {
      const expectedCalls = [aaApiCalls.getUser, aaApiCalls.getUser]
      await withMockFetch(expectedCalls, async () => {
        const { container } = renderView('/')
        await screen.findByAltText('Arlo, by VotingWorks')
        expect(container).toMatchSnapshot()
      })
    })

    it('when logged in as an audit board, shows the login screen', async () => {
      const expectedCalls = [apiMocks.abAuth]
      await withMockFetch(expectedCalls, async () => {
        renderView('/')
        await screen.findByRole('button', { name: 'Log in to your audit' })
      })
    })
  })

  describe('/election/:electionId/audit-board/:auditBoardId', () => {
    it('renders unauthenticated properly', async () => {
      const expectedCalls = [apiMocks.failedAuth]
      await withMockFetch(expectedCalls, async () => {
        const { container } = renderView(
          '/election/1/audit-board/audit-board-1'
        )
        await screen.findByAltText('Arlo, by VotingWorks')
        expect(container).toMatchSnapshot()
      })
    })

    it('renders ja logged in properly', async () => {
      const expectedCalls = [jaApiCalls.getUser]
      await withMockFetch(expectedCalls, async () => {
        const { container } = renderView(
          '/election/1/audit-board/audit-board-1'
        )
        await screen.findByAltText('Arlo, by VotingWorks')
        expect(container).toMatchSnapshot()
      })
    })

    it('renders aa logged in properly', async () => {
      const expectedCalls = [aaApiCalls.getUser, aaApiCalls.getUser]
      await withMockFetch(expectedCalls, async () => {
        const { container } = renderView(
          '/election/1/audit-board/audit-board-1'
        )
        await screen.findByAltText('Arlo, by VotingWorks')
        expect(container).toMatchSnapshot()
      })
    })

    it.skip('renders ab logged in properly', async () => {
      const expectedCalls = [apiMocks.abAuth]
      await withMockFetch(expectedCalls, async () => {
        const { container } = render(
          <MemoryRouter
            initialEntries={['/election/1/audit-board/audit-board-1']}
            initialIndex={0}
          >
            <App />
          </MemoryRouter>
        )
        await screen.findByText('Audit Board #1: Member Sign-in')
        expect(container).toMatchSnapshot()
      })
    })
  })

  describe('/election/:electionId/jurisdiction/:jurisdictionId', () => {
    it('renders unauthenticated properly', async () => {
      const expectedCalls = [apiMocks.failedAuth]
      await withMockFetch(expectedCalls, async () => {
        const { container } = renderView(
          '/election/1/jurisdiction/jurisdiction-id-1'
        )
        await screen.findByAltText('Arlo, by VotingWorks')
        expect(container).toMatchSnapshot()
      })
    })

    it('renders ja logged in properly', async () => {
      const expectedCalls = [
        jaApiCalls.getUser,
        jaApiCalls.getSettings(auditSettings.batchComparisonAll),
        jaApiCalls.getRounds,
        jaApiCalls.getBallotManifestFile(manifestMocks.empty),
        jaApiCalls.getBatchTalliesFile(talliesMocks.empty),
        jaApiCalls.getCVRSfile(cvrsMocks.empty),
      ]
      await withMockFetch(expectedCalls, async () => {
        const { container } = renderView(
          '/election/1/jurisdiction/jurisdiction-id-1'
        )
        await screen.findByAltText('Arlo, by VotingWorks')
        expect(container).toMatchSnapshot()
      })
    })

    it('renders aa logged in properly', async () => {
      const expectedCalls = [aaApiCalls.getUser, aaApiCalls.getUser]
      await withMockFetch(expectedCalls, async () => {
        const { container } = renderView(
          '/election/1/jurisdiction/jurisdiction-id-1'
        )
        await screen.findByText('New Audit')
        expect(container).toMatchSnapshot()
      })
    })
  })
})
