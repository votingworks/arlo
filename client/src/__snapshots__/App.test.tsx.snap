// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App / renders aa logged in properly 1`] = `
<div>
  <div
    class="Toastify"
  />
  <div
    class="sc-cmTdod ftrAGQ"
  >
    <div
      class="bp3-navbar sc-EHOje gMcVig"
    >
      <div
        class="sc-bxivhb sc-bZQynM ezCmGO"
      >
        <div
          class="bp3-navbar-group bp3-align-left"
        >
          <div
            class="bp3-navbar-heading"
          >
            <a
              href="/"
            >
              <img
                alt="Arlo, by VotingWorks"
                src="/arlo.png"
              />
            </a>
          </div>
        </div>
        <div
          class="bp3-navbar-group bp3-align-right"
        >
          <div
            class="sc-ifAKCX jNLPyG"
            id="reset-button-wrapper"
          />
          <button
            class="bp3-button bp3-small sc-bwzfXH eBfcXh"
            type="button"
          >
            <span
              class="bp3-button-text"
            >
              Log out
            </span>
          </button>
        </div>
      </div>
    </div>
    <div
      class="sc-htpNat cBZqYS"
    >
      <div
        class="sc-bxivhb hMnACT"
      >
        <div
          style="width: 50%;"
        >
          <div
            class="sc-ktHwxA bRARec"
          >
            <div>
              <h2>
                Audits - 
                State of California
              </h2>
              <p>
                You haven't created any audits yet for 
                State of California
              </p>
            </div>
          </div>
        </div>
        <div
          style="width: 50%;"
        >
          <div
            class="sc-kafWEX hSEaCR"
          >
            <h2>
              New Audit
            </h2>
            <div
              class="sc-jTzLTM hPHXqn"
            />
            <div
              class="sc-jTzLTM hPHXqn"
            >
              <label
                for="auditName"
              >
                <p>
                  Audit name
                </p>
                <div
                  class="sc-iELTvK dsVrZo sc-kAzzGY bLtYsq"
                >
                  <div
                    class="bp3-input-group sc-chPdSV PhUsl"
                  >
                    <input
                      class="bp3-input"
                      id="auditName"
                      name="auditName"
                      style="padding-right: 10px;"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </label>
            </div>
            <div
              class="sc-jTzLTM hPHXqn"
            >
              <label
                for="auditType"
              >
                <p>
                  Audit type
                </p>
                <div>
                  <label
                    class="bp3-control bp3-radio"
                  >
                    <input
                      checked=""
                      name="auditType"
                      type="radio"
                      value="BALLOT_POLLING"
                    />
                    <span
                      class="bp3-control-indicator"
                    />
                    Ballot Polling
                  </label>
                  <div
                    class="sc-feJyhm jdmEfO"
                  >
                    <label
                      for="auditMathType"
                    >
                      <p>
                        Ballot polling type
                      </p>
                      <div>
                        <label
                          class="bp3-control bp3-radio"
                        >
                          <input
                            checked=""
                            name="auditMathType"
                            type="radio"
                            value="BRAVO"
                          />
                          <span
                            class="bp3-control-indicator"
                          />
                          BRAVO
                        </label>
                        <label
                          class="bp3-control bp3-radio"
                        >
                          <input
                            name="auditMathType"
                            type="radio"
                            value="MINERVA"
                          />
                          <span
                            class="bp3-control-indicator"
                          />
                          Minerva (Not recommended)
                        </label>
                      </div>
                    </label>
                  </div>
                  <label
                    class="bp3-control bp3-radio"
                  >
                    <input
                      name="auditType"
                      type="radio"
                      value="BATCH_COMPARISON"
                    />
                    <span
                      class="bp3-control-indicator"
                    />
                    Batch Comparison
                  </label>
                  <label
                    class="bp3-control bp3-radio"
                  >
                    <input
                      name="auditType"
                      type="radio"
                      value="BALLOT_COMPARISON"
                    />
                    <span
                      class="bp3-control-indicator"
                    />
                    Ballot Comparison
                  </label>
                </div>
              </label>
            </div>
            <button
              class="bp3-button bp3-fill bp3-large bp3-intent-primary sc-bwzfXH eBfcXh"
              type="button"
            >
              <span
                class="bp3-button-text"
              >
                Create Audit
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`App / renders ja logged in properly 1`] = `
<div>
  <div
    class="Toastify"
  />
  <div
    class="sc-cmTdod ftrAGQ"
  >
    <div
      class="bp3-navbar sc-EHOje gMcVig"
    >
      <div
        class="sc-bxivhb sc-bZQynM ezCmGO"
      >
        <div
          class="bp3-navbar-group bp3-align-left"
        >
          <div
            class="bp3-navbar-heading"
          >
            <a
              href="/"
            >
              <img
                alt="Arlo, by VotingWorks"
                src="/arlo.png"
              />
            </a>
          </div>
          <div
            class="bp3-navbar-heading"
          />
        </div>
        <div
          class="bp3-navbar-group bp3-align-right"
        >
          <div
            class="sc-ifAKCX jNLPyG"
            id="reset-button-wrapper"
          />
          <button
            class="bp3-button bp3-small sc-bwzfXH eBfcXh"
            type="button"
          >
            <span
              class="bp3-button-text"
            >
              Log out
            </span>
          </button>
        </div>
      </div>
    </div>
    <div
      class="sc-htpNat cBZqYS"
    >
      <div
        class="sc-bxivhb hMnACT"
      >
        <div
          style="width: 50%;"
        >
          <div
            class="sc-ktHwxA bRARec"
          >
            <div>
              <h2>
                Jurisdictions - 
                audit one
              </h2>
              <button
                class="bp3-button bp3-fill bp3-large bp3-intent-primary sc-cIShpX iRnxMq"
                href="/election/1/jurisdiction/jurisdiction-id-1"
                type="button"
              >
                <span
                  class="bp3-button-text"
                >
                  Jurisdiction One
                </span>
              </button>
              <button
                class="bp3-button bp3-fill bp3-large bp3-intent-primary sc-cIShpX iRnxMq"
                href="/election/1/jurisdiction/jurisdiction-id-3"
                type="button"
              >
                <span
                  class="bp3-button-text"
                >
                  Jurisdiction Three
                </span>
              </button>
            </div>
            <div>
              <h2>
                Jurisdictions - 
                audit two
              </h2>
              <button
                class="bp3-button bp3-fill bp3-large bp3-intent-primary sc-cIShpX iRnxMq"
                href="/election/2/jurisdiction/jurisdiction-id-2"
                type="button"
              >
                <span
                  class="bp3-button-text"
                >
                  Jurisdiction Two
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`App / renders unauthenticated properly 1`] = `
<div>
  <div
    class="Toastify"
  />
  <div
    class="sc-cmTdod ftrAGQ"
  >
    <div
      class="bp3-navbar sc-EHOje gMcVig"
    >
      <div
        class="sc-bxivhb sc-bZQynM ezCmGO"
      >
        <div
          class="bp3-navbar-group bp3-align-left"
        >
          <div
            class="bp3-navbar-heading"
          >
            <a
              href="/"
            >
              <img
                alt="Arlo, by VotingWorks"
                src="/arlo.png"
              />
            </a>
          </div>
        </div>
        <div
          class="bp3-navbar-group bp3-align-right"
        >
          <div
            class="sc-ifAKCX jNLPyG"
            id="reset-button-wrapper"
          />
        </div>
      </div>
    </div>
    <div
      class="sc-hEsumM cWAYk"
    >
      <img
        alt="Arlo, by VotingWorks"
        height="50px"
        src="/arlo.png"
      />
      <div
        class="bp3-card bp3-elevation-0"
        style="margin: 25px 0px 15px 0px;"
      >
        <p>
          Participating in an audit in your local jurisdiction?
        </p>
        <a
          class="bp3-button bp3-large bp3-intent-primary"
          href="/auth/jurisdictionadmin/start"
          role="button"
          tabindex="0"
        >
          <span
            class="bp3-button-text"
          >
            Log in to your audit
          </span>
        </a>
      </div>
      <div>
        <p>
          State-level audit administrators:
           
          <a
            href="/auth/auditadmin/start"
          >
            Log in as an admin
          </a>
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`App /election/:electionId/audit-board/:auditBoardId renders aa logged in properly 1`] = `
<div>
  <div
    class="Toastify"
  />
  <div
    class="sc-cmTdod ftrAGQ"
  >
    <div
      class="bp3-navbar sc-EHOje gMcVig"
    >
      <div
        class="sc-bxivhb sc-bZQynM ezCmGO"
      >
        <div
          class="bp3-navbar-group bp3-align-left"
        >
          <div
            class="bp3-navbar-heading"
          >
            <a
              href="/"
            >
              <img
                alt="Arlo, by VotingWorks"
                src="/arlo.png"
              />
            </a>
          </div>
        </div>
        <div
          class="bp3-navbar-group bp3-align-right"
        >
          <div
            class="sc-ifAKCX jNLPyG"
            id="reset-button-wrapper"
          />
          <button
            class="bp3-button bp3-small sc-bwzfXH eBfcXh"
            type="button"
          >
            <span
              class="bp3-button-text"
            >
              Log out
            </span>
          </button>
        </div>
      </div>
    </div>
    <div
      class="sc-htpNat cBZqYS"
    >
      <div
        class="sc-bxivhb hMnACT"
      >
        <div
          style="width: 50%;"
        >
          <div
            class="sc-ktHwxA bRARec"
          >
            <div>
              <h2>
                Audits - 
                State of California
              </h2>
              <p>
                You haven't created any audits yet for 
                State of California
              </p>
            </div>
          </div>
        </div>
        <div
          style="width: 50%;"
        >
          <div
            class="sc-kafWEX hSEaCR"
          >
            <h2>
              New Audit
            </h2>
            <div
              class="sc-jTzLTM hPHXqn"
            />
            <div
              class="sc-jTzLTM hPHXqn"
            >
              <label
                for="auditName"
              >
                <p>
                  Audit name
                </p>
                <div
                  class="sc-iELTvK dsVrZo sc-kAzzGY bLtYsq"
                >
                  <div
                    class="bp3-input-group sc-chPdSV PhUsl"
                  >
                    <input
                      class="bp3-input"
                      id="auditName"
                      name="auditName"
                      style="padding-right: 10px;"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </label>
            </div>
            <div
              class="sc-jTzLTM hPHXqn"
            >
              <label
                for="auditType"
              >
                <p>
                  Audit type
                </p>
                <div>
                  <label
                    class="bp3-control bp3-radio"
                  >
                    <input
                      checked=""
                      name="auditType"
                      type="radio"
                      value="BALLOT_POLLING"
                    />
                    <span
                      class="bp3-control-indicator"
                    />
                    Ballot Polling
                  </label>
                  <div
                    class="sc-feJyhm jdmEfO"
                  >
                    <label
                      for="auditMathType"
                    >
                      <p>
                        Ballot polling type
                      </p>
                      <div>
                        <label
                          class="bp3-control bp3-radio"
                        >
                          <input
                            checked=""
                            name="auditMathType"
                            type="radio"
                            value="BRAVO"
                          />
                          <span
                            class="bp3-control-indicator"
                          />
                          BRAVO
                        </label>
                        <label
                          class="bp3-control bp3-radio"
                        >
                          <input
                            name="auditMathType"
                            type="radio"
                            value="MINERVA"
                          />
                          <span
                            class="bp3-control-indicator"
                          />
                          Minerva (Not recommended)
                        </label>
                      </div>
                    </label>
                  </div>
                  <label
                    class="bp3-control bp3-radio"
                  >
                    <input
                      name="auditType"
                      type="radio"
                      value="BATCH_COMPARISON"
                    />
                    <span
                      class="bp3-control-indicator"
                    />
                    Batch Comparison
                  </label>
                  <label
                    class="bp3-control bp3-radio"
                  >
                    <input
                      name="auditType"
                      type="radio"
                      value="BALLOT_COMPARISON"
                    />
                    <span
                      class="bp3-control-indicator"
                    />
                    Ballot Comparison
                  </label>
                </div>
              </label>
            </div>
            <button
              class="bp3-button bp3-fill bp3-large bp3-intent-primary sc-bwzfXH eBfcXh"
              type="button"
            >
              <span
                class="bp3-button-text"
              >
                Create Audit
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`App /election/:electionId/audit-board/:auditBoardId renders ab logged in properly 1`] = `<div />`;

exports[`App /election/:electionId/audit-board/:auditBoardId renders ja logged in properly 1`] = `
<div>
  <div
    class="Toastify"
  />
  <div
    class="sc-cmTdod ftrAGQ"
  >
    <div
      class="bp3-navbar sc-EHOje gMcVig"
    >
      <div
        class="sc-bxivhb sc-bZQynM ezCmGO"
      >
        <div
          class="bp3-navbar-group bp3-align-left"
        >
          <div
            class="bp3-navbar-heading"
          >
            <a
              href="/"
            >
              <img
                alt="Arlo, by VotingWorks"
                src="/arlo.png"
              />
            </a>
          </div>
          <div
            class="bp3-navbar-heading"
          />
        </div>
        <div
          class="bp3-navbar-group bp3-align-right"
        >
          <div
            class="sc-ifAKCX jNLPyG"
            id="reset-button-wrapper"
          />
          <button
            class="bp3-button bp3-small sc-bwzfXH eBfcXh"
            type="button"
          >
            <span
              class="bp3-button-text"
            >
              Log out
            </span>
          </button>
        </div>
      </div>
    </div>
    <div
      class="sc-htpNat cBZqYS"
    >
      <div
        class="sc-bxivhb hMnACT"
      >
        <div
          style="width: 50%;"
        >
          <div
            class="sc-ktHwxA bRARec"
          >
            <div>
              <h2>
                Jurisdictions - 
                audit one
              </h2>
              <button
                class="bp3-button bp3-fill bp3-large bp3-intent-primary sc-cIShpX iRnxMq"
                href="/election/1/jurisdiction/jurisdiction-id-1"
                type="button"
              >
                <span
                  class="bp3-button-text"
                >
                  Jurisdiction One
                </span>
              </button>
              <button
                class="bp3-button bp3-fill bp3-large bp3-intent-primary sc-cIShpX iRnxMq"
                href="/election/1/jurisdiction/jurisdiction-id-3"
                type="button"
              >
                <span
                  class="bp3-button-text"
                >
                  Jurisdiction Three
                </span>
              </button>
            </div>
            <div>
              <h2>
                Jurisdictions - 
                audit two
              </h2>
              <button
                class="bp3-button bp3-fill bp3-large bp3-intent-primary sc-cIShpX iRnxMq"
                href="/election/2/jurisdiction/jurisdiction-id-2"
                type="button"
              >
                <span
                  class="bp3-button-text"
                >
                  Jurisdiction Two
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`App /election/:electionId/audit-board/:auditBoardId renders unauthenticated properly 1`] = `
<div>
  <div
    class="Toastify"
  />
  <div
    class="sc-cmTdod ftrAGQ"
  >
    <div
      class="bp3-navbar sc-EHOje gMcVig"
    >
      <div
        class="sc-bxivhb sc-bZQynM ezCmGO"
      >
        <div
          class="bp3-navbar-group bp3-align-left"
        >
          <div
            class="bp3-navbar-heading"
          >
            <a
              href="/"
            >
              <img
                alt="Arlo, by VotingWorks"
                src="/arlo.png"
              />
            </a>
          </div>
        </div>
        <div
          class="bp3-navbar-group bp3-align-right"
        >
          <div
            class="sc-ifAKCX jNLPyG"
            id="reset-button-wrapper"
          />
        </div>
      </div>
    </div>
    <div
      class="sc-hEsumM cWAYk"
    >
      <img
        alt="Arlo, by VotingWorks"
        height="50px"
        src="/arlo.png"
      />
      <div
        class="bp3-card bp3-elevation-0"
        style="margin: 25px 0px 15px 0px;"
      >
        <p>
          Participating in an audit in your local jurisdiction?
        </p>
        <a
          class="bp3-button bp3-large bp3-intent-primary"
          href="/auth/jurisdictionadmin/start"
          role="button"
          tabindex="0"
        >
          <span
            class="bp3-button-text"
          >
            Log in to your audit
          </span>
        </a>
      </div>
      <div>
        <p>
          State-level audit administrators:
           
          <a
            href="/auth/auditadmin/start"
          >
            Log in as an admin
          </a>
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`App /election/:electionId/jurisdiction/:jurisdictionId renders aa logged in properly 1`] = `
<div>
  <div
    class="Toastify"
  />
  <div
    class="sc-cmTdod ftrAGQ"
  >
    <div
      class="bp3-navbar sc-EHOje gMcVig"
    >
      <div
        class="sc-bxivhb sc-bZQynM ezCmGO"
      >
        <div
          class="bp3-navbar-group bp3-align-left"
        >
          <div
            class="bp3-navbar-heading"
          >
            <a
              href="/"
            >
              <img
                alt="Arlo, by VotingWorks"
                src="/arlo.png"
              />
            </a>
          </div>
        </div>
        <div
          class="bp3-navbar-group bp3-align-right"
        >
          <div
            class="sc-ifAKCX jNLPyG"
            id="reset-button-wrapper"
          />
          <button
            class="bp3-button bp3-small sc-bwzfXH eBfcXh"
            type="button"
          >
            <span
              class="bp3-button-text"
            >
              Log out
            </span>
          </button>
        </div>
      </div>
    </div>
    <div
      class="sc-htpNat cBZqYS"
    >
      <div
        class="sc-bxivhb hMnACT"
      >
        <div
          style="width: 50%;"
        >
          <div
            class="sc-ktHwxA bRARec"
          >
            <div>
              <h2>
                Audits - 
                State of California
              </h2>
              <p>
                You haven't created any audits yet for 
                State of California
              </p>
            </div>
          </div>
        </div>
        <div
          style="width: 50%;"
        >
          <div
            class="sc-kafWEX hSEaCR"
          >
            <h2>
              New Audit
            </h2>
            <div
              class="sc-jTzLTM hPHXqn"
            />
            <div
              class="sc-jTzLTM hPHXqn"
            >
              <label
                for="auditName"
              >
                <p>
                  Audit name
                </p>
                <div
                  class="sc-iELTvK dsVrZo sc-kAzzGY bLtYsq"
                >
                  <div
                    class="bp3-input-group sc-chPdSV PhUsl"
                  >
                    <input
                      class="bp3-input"
                      id="auditName"
                      name="auditName"
                      style="padding-right: 10px;"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </label>
            </div>
            <div
              class="sc-jTzLTM hPHXqn"
            >
              <label
                for="auditType"
              >
                <p>
                  Audit type
                </p>
                <div>
                  <label
                    class="bp3-control bp3-radio"
                  >
                    <input
                      checked=""
                      name="auditType"
                      type="radio"
                      value="BALLOT_POLLING"
                    />
                    <span
                      class="bp3-control-indicator"
                    />
                    Ballot Polling
                  </label>
                  <div
                    class="sc-feJyhm jdmEfO"
                  >
                    <label
                      for="auditMathType"
                    >
                      <p>
                        Ballot polling type
                      </p>
                      <div>
                        <label
                          class="bp3-control bp3-radio"
                        >
                          <input
                            checked=""
                            name="auditMathType"
                            type="radio"
                            value="BRAVO"
                          />
                          <span
                            class="bp3-control-indicator"
                          />
                          BRAVO
                        </label>
                        <label
                          class="bp3-control bp3-radio"
                        >
                          <input
                            name="auditMathType"
                            type="radio"
                            value="MINERVA"
                          />
                          <span
                            class="bp3-control-indicator"
                          />
                          Minerva (Not recommended)
                        </label>
                      </div>
                    </label>
                  </div>
                  <label
                    class="bp3-control bp3-radio"
                  >
                    <input
                      name="auditType"
                      type="radio"
                      value="BATCH_COMPARISON"
                    />
                    <span
                      class="bp3-control-indicator"
                    />
                    Batch Comparison
                  </label>
                  <label
                    class="bp3-control bp3-radio"
                  >
                    <input
                      name="auditType"
                      type="radio"
                      value="BALLOT_COMPARISON"
                    />
                    <span
                      class="bp3-control-indicator"
                    />
                    Ballot Comparison
                  </label>
                </div>
              </label>
            </div>
            <button
              class="bp3-button bp3-fill bp3-large bp3-intent-primary sc-bwzfXH eBfcXh"
              type="button"
            >
              <span
                class="bp3-button-text"
              >
                Create Audit
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`App /election/:electionId/jurisdiction/:jurisdictionId renders ja logged in properly 1`] = `
<div>
  <div
    class="Toastify"
  />
  <div
    class="sc-cmTdod ftrAGQ"
  >
    <div
      class="bp3-navbar sc-EHOje gMcVig"
    >
      <div
        class="sc-bxivhb sc-bZQynM ezCmGO"
      >
        <div
          class="bp3-navbar-group bp3-align-left"
        >
          <div
            class="bp3-navbar-heading"
          >
            <a
              href="/"
            >
              <img
                alt="Arlo, by VotingWorks"
                src="/arlo.png"
              />
            </a>
          </div>
          <div
            class="bp3-navbar-heading"
          >
            Jurisdiction: Jurisdiction One
          </div>
        </div>
        <div
          class="bp3-navbar-group bp3-align-right"
        >
          <div
            class="sc-ifAKCX jNLPyG"
            id="reset-button-wrapper"
          />
          <button
            class="bp3-button bp3-small sc-bwzfXH eBfcXh"
            type="button"
          >
            <span
              class="bp3-button-text"
            >
              Log out
            </span>
          </button>
        </div>
      </div>
    </div>
    <div
      class="sc-htpNat cBZqYS"
    >
      <div
        class="bp3-callout sc-brqgnP jodPLw"
      >
        <div
          class="sc-bxivhb hMnACT"
        >
          <div
            class="text"
          >
            <h4
              class="bp3-heading"
            >
              The audit has not started.
            </h4>
            <p>
              0/2 files successfully uploaded.
            </p>
          </div>
        </div>
      </div>
      <div
        class="sc-bxivhb sc-TOsTZ dNNkJG"
      >
        <h2
          class="bp3-heading sc-gzVnrw hQaoin"
        >
          Audit Source Data
        </h2>
        <form>
          <div
            class="sc-dnqmqq kymPuI"
          >
            <h2
              class="bp3-heading sc-gzVnrw hQaoin"
            />
            <div
              class="sc-jTzLTM hPHXqn"
            >
              <h4
                class="bp3-heading"
              >
                Ballot Manifest
              </h4>
              <div
                class="sc-fjdhpX kBSvWX"
              >
                Click "Browse" to choose the appropriate Ballot Manifest file from your computer. This file should be a comma-separated list of all the ballot boxes/containers used to store ballots for this particular election, plus a count of how many ballot cards (individual pieces of paper) are stored in each container.
                <br />
                <br />
                <a
                  href="/sample_ballot_manifest.csv"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  (Click here to view a sample file in the correct format.)
                </a>
              </div>
            </div>
            <div
              class="sc-jTzLTM hPHXqn"
            >
              <label
                class="bp3-file-input"
              >
                <input
                  accept=".csv"
                  name="csv"
                  type="file"
                />
                <span
                  class="bp3-file-upload-input"
                >
                  Select a CSV...
                </span>
              </label>
            </div>
            <div>
              <button
                class="bp3-button bp3-intent-primary sc-bwzfXH eBfcXh"
                type="submit"
              >
                <span
                  class="bp3-button-text"
                >
                  Upload File
                </span>
              </button>
            </div>
          </div>
        </form>
        <form>
          <div
            class="sc-dnqmqq kymPuI"
          >
            <h2
              class="bp3-heading sc-gzVnrw hQaoin"
            />
            <div
              class="sc-jTzLTM hPHXqn"
            >
              <h4
                class="bp3-heading"
              >
                Candidate Totals by Batch
              </h4>
              <div
                class="sc-fjdhpX kBSvWX"
              >
                Click "Browse" to choose the appropriate Candidate Totals by Batch file from your computer. This file should be a comma-separated list of all the ballot boxes/containers used to store ballots for this particular election, plus a count of how many votes were counted for each candidate in each of those containers.
                <br />
                <br />
                <a
                  href="/sample_candidate_totals_by_batch.csv"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  (Click here to view a sample file in the correct format.)
                </a>
              </div>
            </div>
            <div
              class="sc-jTzLTM hPHXqn"
            >
              <label
                class="bp3-file-input bp3-disabled"
              >
                <input
                  accept=".csv"
                  disabled=""
                  name="csv"
                  type="file"
                />
                <span
                  class="bp3-file-upload-input"
                >
                  Select a CSV...
                </span>
              </label>
            </div>
            <div>
              <button
                class="bp3-button bp3-disabled bp3-intent-primary sc-bwzfXH eBfcXh"
                disabled=""
                tabindex="-1"
                type="submit"
              >
                <span
                  class="bp3-button-text"
                >
                  Upload File
                </span>
              </button>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;

exports[`App /election/:electionId/jurisdiction/:jurisdictionId renders unauthenticated properly 1`] = `
<div>
  <div
    class="Toastify"
  />
  <div
    class="sc-cmTdod ftrAGQ"
  >
    <div
      class="bp3-navbar sc-EHOje gMcVig"
    >
      <div
        class="sc-bxivhb sc-bZQynM ezCmGO"
      >
        <div
          class="bp3-navbar-group bp3-align-left"
        >
          <div
            class="bp3-navbar-heading"
          >
            <a
              href="/"
            >
              <img
                alt="Arlo, by VotingWorks"
                src="/arlo.png"
              />
            </a>
          </div>
        </div>
        <div
          class="bp3-navbar-group bp3-align-right"
        >
          <div
            class="sc-ifAKCX jNLPyG"
            id="reset-button-wrapper"
          />
        </div>
      </div>
    </div>
    <div
      class="sc-hEsumM cWAYk"
    >
      <img
        alt="Arlo, by VotingWorks"
        height="50px"
        src="/arlo.png"
      />
      <div
        class="bp3-card bp3-elevation-0"
        style="margin: 25px 0px 15px 0px;"
      >
        <p>
          Participating in an audit in your local jurisdiction?
        </p>
        <a
          class="bp3-button bp3-large bp3-intent-primary"
          href="/auth/jurisdictionadmin/start"
          role="button"
          tabindex="0"
        >
          <span
            class="bp3-button-text"
          >
            Log in to your audit
          </span>
        </a>
      </div>
      <div>
        <p>
          State-level audit administrators:
           
          <a
            href="/auth/auditadmin/start"
          >
            Log in as an admin
          </a>
        </p>
      </div>
    </div>
  </div>
</div>
`;
